src/functions.js

```diff
@@ -0,0 +1,94 @@
+class Task {
+  constructor(i, c = false, d) {
+    this.index = i;
+    this.completed = c;
+    this.description = d;
+  }
+}
+
+const chkboxStatus = (item) => item.checked;
+
+export default chkboxStatus;
+
+let myList = [];
+
+const saveStorage = () => {
+  localStorage.setItem('tasks', JSON.stringify(myList));
+};
+
+const addLi = (ul, index, completed, description) => {
+  const task = document.createElement('li');
+  const cBox = document.createElement('input');
+  cBox.type = 'checkbox';
+  cBox.id = index;
+  cBox.checked = completed;
+  cBox.addEventListener('change', () => {
+    if (chkboxStatus(cBox)) {
+      myList[parseInt(cBox.id, 10)].completed = true;
+    } else {
+      myList[parseInt(cBox.id, 10)].completed = false;
+    }
+    saveStorage();
+  });
+  const label = document.createElement('label');
+  label.contentEditable = true;
+  label.innerText = description;
+  label.addEventListener('focusout', (e) => {
+    label.parentNode.style.backgroundColor = 'initial';
+    label.parentNode.childNodes[2].childNodes[0].style.display = 'initial';
+    label.parentNode.childNodes[2].childNodes[2].style.display = 'none';
+    myList[parseInt(cBox.id, 10)].description = e.composedPath()[0].innerText;
+    saveStorage();
+    setTimeout(() => {
+      label.parentNode.childNodes[2].style.pointerEvents = 'none';
+    }, 500);
+  });
+  const elBtn = document.createElement('button');
+  elBtn.className = 'element-button';
+  elBtn.addEventListener('click', (e) => {
+    // eslint-disable-next-line no-use-before-define
+    Delete(ul, e.currentTarget.parentNode.childNodes[0].id);
+    e.target.closest('li').remove();
+  }, false);
```

Comment:
- [ ] It looks like this method has some bugs here. When I try to delete task 2 it deletes Task 3 instead. Kindly fix this bug. :pray: 

![Screenshot from 2021-10-29 22-17-16](https://user-images.githubusercontent.com/64894927/139490576-39b223d4-97d2-49c6-bee3-ab654cd2f885.png)

While deleting Task2
![Screenshot from 2021-10-29 22-17-20](https://user-images.githubusercontent.com/64894927/139490607-3cfbcae1-9058-4db8-a977-10a88025f6c3.png)

After deleting